docroot/modules/custom/sitenow_media_wysiwyg/sitenow_media_wysiwyg.module

```diff
@@ -35,6 +35,57 @@ function sitenow_media_wysiwyg_inline_entity_form_entity_form_alter(&$entity_for
   }
 }
 
+/**
+ * Implements hook_entity_presave().
+ *
+ * Ensures Panopto media entities have their titles set from the video's title.
+ */
+function sitenow_media_wysiwyg_entity_presave($entity) {
+  // Only process media entities.
+  if ($entity->getEntityTypeId() !== 'media') {
+    return;
+  }
+
+  // Only process Panopto media.
+  if ($entity->bundle() !== 'panopto') {
+    return;
+  }
+
+  // Get the source field configuration.
+  $source = $entity->getSource();
+  $source_field = $source->getConfiguration()['source_field'];
+
+  // Only proceed if we have a source field value.
+  if (!$entity->hasField($source_field) || $entity->get($source_field)->isEmpty()) {
+    return;
+  }
+
+  // Get the source field value.
+  $source_field_value = $entity->get($source_field)->getValue();
+
+  // If we have a title in the source field, use it for the media entity name.
```

Comment:
This is not what we want to do. We want to set the source field's `title` value with the html title we get from the Guzzle request. Since this fires for both initial save and updates after, might as well check for a updated value from the source (changed video title) and use it. The media name name is set in the Panopto file and we want to keep that there.